machine.hpp

```diff
@@ -1,114 +0,0 @@
-/*
- * machine.hpp
- *
- * Baker, Ballard, Jager-Kujawa
- * CSC 341
- * Spring 2016
- */
-
-#ifndef MACHINE
-#define MACHINE
-
-#include <string>
-#include <iostream>
-#include <iomanip>
-	using namespace std;
-
-// Debugging flags
-#define DEBUG
-#define DEBUG_VERBOSE
-#undef DEBUG_VERBOSE
-//undef DEBUG
-
-
-// If DEBUG_VERBOSE, then DEBUG
-#ifdef DEBUG_VERBOSE
-	#ifndef DEBUG
-		#define DEBUG
-	#endif
-#endif
-
-/* Define condition codes for hardware functions
-> : 001
-< : 100
-= : 010
-!= : 101
->= : 011
-<= : 110
-*/
-#define GRT 1
-#define LST 4
-#define EQL 2
-#define NEQ 5
-#define GEQ 3
-#define LEQ 6
-
-// Define addressing modes
-#define DIRECT 0
-#define IMMEDIATE 1
-
-// Define machine registers type
-// Machine register type also has a condition code register
-typedef struct {
-	unsigned short int r1, r2, r3, rA, IR, PC:8;
-	unsigned short int CR:3;
-} registers;
```

Comment:
Let's add a **Page Table Base Register**. This would essentially be a pointer to the page table object. 

Because the process control block is essentially just a collection of registers, we can use the registers struct to represent our PCBs. Adding in the PTBR satisfies the PCB requirement of keeping a pointer to that process' page table.

Furthermore, because we are also representing our machine registers as a registers struct, when copying the process registers into the machine registers, the machine PTBR would also be reassigned to point to the now-running process.

**Ultimately we're going to have an MMU method/object/something-or-other *in hardware* that will use the PTBR to perform the logical-to-physical address translation.**

* If PTBR is null, should the MMU throw an error? Or should it simply return the logical address, untranslated?